packs/caddy/templates/caddy.nomad.tpl

```diff
@@ -0,0 +1,85 @@
+job [[ template "job_name" . ]] {
+[[ template "region" . ]]
+datacenters = [ [[ range $idx, $dc := .caddy.datacenters ]][[if $idx]],[[end]][[ $dc | quote ]][[ end ]] ]
+  namespace = [[ .caddy.namespace | quote ]]
+
+  type = "system"
+
+  // must have linux for network mode
+  constraint {
+    attribute = "${attr.kernel.name}"
+    value     = "linux"
+  }
+
+  group "caddy" {
+    count = 1
+
+    # Using ephemeral_disk to store Caddy data.
+    # The data directory must not be treated as a cache.
+    # Caddy stores TLS certificates, private keys, OCSP staples, and other necessary information to the data directory.
+    # The path to this directory is configured via `XDG_DATA_HOME` variable.
+    ephemeral_disk {
+      migrate = true
+      size    = 300
+      sticky  = true
+    }
+
+    network {
+      port "https" {
+        static = [[ .caddy.https_port ]]
+        to     = 443
+      }
+      port "http" {
+        static = [[ .caddy.http_port ]]
+        to     = 80
+      }
+    }
```

Comment:
Can we allow more operator control over this including the mode which i'd like to default to bridge? The [Prometheus pack](https://github.com/hashicorp/nomad-pack-community-registry/blob/main/packs/prometheus/templates/prometheus.nomad.tpl#L18-L25) includes an example.